adafruit_rgbled.py

```diff
@@ -90,10 +80,25 @@ class RGBLED:
         import adafruit_rgbled
         with adafruit_rgbled.RGBLED(board.D5, board.D6, board.D7, invert_pwm=True) as rgb_led:
             rgb_led.color = (0, 255, 0)
-
     """
 
-    def __init__(self, red_pin, green_pin, blue_pin, invert_pwm=False):
+    def __init__(
+        self,
+        red_pin: Union["microcontroller.Pin", PWMOut, "PWMChannel"],
```

Comment:
Is there any reason you're using quotes instead of importing within the `try`/`except` block?

Reply:
For the "microcontroller.Pin" I was getting "Not Implemented" exceptions in my PyCharm environment. That now happens with the `from pwmio import PWMOut` which makes this moot. I've changed this to use `from microcontroller import Pin`

For the "PWMChannel" I couldn't find a library with `PWMChannel` that I can import on my laptop. The `#circuitpython-dev` Discord channel suggested using quotes. If you know of a library that I can use, I'm happy to make that change too.

Reply:
…and the CI is failing with 
```
  File "/home/runner/work/Adafruit_CircuitPython_RGBLED/Adafruit_CircuitPython_RGBLED/adafruit_rgbled.py", line 98, in RGBLED
    red_pin: Union[microcontroller.Pin, PWMOut, "PWMChannel"],
                   ^^^^^^^^^^^^^^^
NameError: name 'microcontroller' is not defined

Error: Process completed with exit code 2.
```

Which may be why I used the quotes. Can't remember TBH!

Reply:
It looks like it's failing because it isn't actually imported in the file.  If you add it as an import to the `try`/`except` block, it should work fine without quotes.  Same thing for `PWMChannel`.

Reply:
With import statements in the `try/except` block I can get `Pin` to pass CI but not `PWMChannel`.

```
try:
    from typing import Union, Optional, Type
    from types import TracebackType
    from microcontroller import Pin
    from adafruit_pca9685 import PWMChannel
    from circuitpython_typing.led import ColorBasedColorUnion
except ImportError:
    pass
...
def __init__(
    self,
    red_pin: Union[Pin, PWMOut, PWMChannel],
    green_pin: Union[Pin, PWMOut, PWMChannel],
    blue_pin: Union[Pin, PWMOut, PWMChannel],
    invert_pwm: bool = False,
) -> None:

```

I've added `adafruit-pca9685` to `requirements.txt` but this did not help.

```
Warning, treated as error:
autodoc: failed to import module 'adafruit_rgbled'; the following exception was raised:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/site-packages/sphinx/ext/autodoc/importer.py", line 60, in import_module
    return importlib.import_module(modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/runner/work/Adafruit_CircuitPython_RGBLED/Adafruit_CircuitPython_RGBLED/adafruit_rgbled.py", line 36, in <module>
    class RGBLED:
  File "/home/runner/work/Adafruit_CircuitPython_RGBLED/Adafruit_CircuitPython_RGBLED/adafruit_rgbled.py", line 100, in RGBLED
    red_pin: Union[Pin, PWMOut, PWMChannel],
                                ^^^^^^^^^^
NameError: name 'PWMChannel' is not defined

Error: Process completed with exit code 2.```